internal/cluster/distribution/eks/eksprovider/driver/cluster_creator.go

```diff
@@ -333,6 +334,15 @@ func (c *EksClusterCreator) validate(r *pkgCluster.CreateClusterRequest, logger
 		return errors.WrapIf(err, "failed to create AWS session")
 	}
 
+	if r.Properties.CreateClusterEKS.IAM.ClusterRoleID != "" {
+		iamSession := iam.New(awsSession)
```

Comment:
As per our recent validation efforts, in API requests we only do formal validation.

Reply:
Márk is referring to [this](https://github.com/banzaicloud/adr/blob/master/adrs/pipeline/0006-API-input-validation-user-feedback.md) ADR.

I think we should execute this validation during the cluster creation workflow asynchronously due to such a validation requiring additional AWS calls - in according to what we discussed about the validation kinds and user feedback relations and subsequently the validation location.

I would propose implementing a `GetIAMRole` activity in the `distribution/eks/provider/workflow` package to retrieve the role corresponding to the cluster role ID, execute it in the workflows where such a check is relevant - such as `CreateInfraWorkflow` - and fail the enclosing workflow in case no such role is found.

Reply:
thx 👍  I updated the PR